src/main.rs

```diff
@@ -51,6 +54,31 @@ async fn main() -> Result<()> {
                 ReconstructSource::File { file: _ } => todo!(),
             }
         }
+        Command::Download {
+            http_url,
+            start_block,
+            block_step: _,
+            block_count,
+            file,
+        } => {
+            let fetcher = L1Fetcher::new(&http_url)?;
+            let processor = JsonSerializationProcessor::new(Path::new(&file))?;
+            let (tx, rx) = mpsc::channel::<Vec<CommitBlockInfoV1>>(5);
+
+            tokio::spawn(async move {
+                processor.run(rx).await;
+            });
+
+            let end_block = match block_count {
+                Some(n) => Some(U64([start_block + n])),
+                None => None,
+            };
```

Comment:
```suggestion
            let end_block = block_count.map(|n| U64([start_block + n]));
```
You can use `Option::map` if you only want to alter a `Some` value.